base/compiler/typeinfer.jl

```diff
@@ -148,7 +147,8 @@ function cache_result!(interp::AbstractInterpreter, result::InferenceResult, min
 
     # TODO: also don't store inferred code if we've previously decided to interpret this function
     if !already_inferred
-        code_cache(interp)[result.linfo] = CodeInstance(result, min_valid, max_valid)
+        code_cache(interp)[result.linfo] = CodeInstance(result, min_valid, max_valid,
+            may_compress(interp), may_discard_trees(interp))
```

Comment:
Seems like these got accidentally added from a different commit, but otherwise seems okay

Reply:
Yeah, these are useful to have the remark emitting AbstractInterpreter force results to be kept such that it can correlate the remarks with the original source, but you're right, it's conceptually separate.

Reply:
Ah, maybe just needs a separate commit then, or at least mention in the commit message, so we don't drop this later?

I'd also perhaps have preferred we just added these to the existing configuration set for this (InferenceParams), rather than creating a duplicate strategy with dispatch.